src-ghc/Pact/Server/Server.hs

```diff
@@ -72,8 +72,8 @@ usage =
 serve :: FilePath -> IO ()
 serve configFile = do
   (runServer, asyncCmd, asyncHist) <- setupServer configFile
-  link asyncCmd   -- Must be individually killed with uninterruptibleCancel
-  link asyncHist  -- Must be individually killed with uninterruptibleCancel
+  link asyncCmd
+  link asyncHist
```

Comment:
We should probably leave in those notes unless someone said otherwise

Reply:
Yes is this no longer true? @LindaOrtega you're the expert

Reply:
Combined this function with the setup function. I also kept the comment with a clarifying comment in the latest commit.

Reply:
The comment is no longer needed if we assume (or enforce) that these two asyncs are created in the same thread as `runServer` is called. Because, by calling asyncCancel on the thread running `runServer`, then that thread will be killed and so will its child ones (i.e. asyncCmd and asyncHist). The comment was originally motivated because the hsuite thread was the parent thread of `cmd` and `hist` in the pact continuation tests (no longer the case, see below paragraph). So these two threads had to be manually killed.

I originally split `serve` into two functions because I was getting weird async errors so I wanted to make sure that each async was getting cancelled in the pact continuation tests (way back when). This separation introduced another bug that was revealed during this PR because the server, cmd, and hist threads weren't being linked correctly. So I changed around how the tests created the server so linking was done correctly and the correct thread-parentage was established. So `serve` no longer needed to be split into two functions.